src/NHibernate/Type/GenericOrderedSetType.cs

```diff
@@ -24,9 +27,23 @@ public GenericOrderedSetType(string role, string propertyRef)
 		{
 		}
 
+		static ConstructorInfo GetSetConstructor()
+		{
+			var type = System.Type.GetType("Iesi.Collections.Generic.LinkedHashSet`1, Iesi.Collections", false);
+			if (type == null)
+				throw new TypeLoadException(
+					@"Could not load Iesi.Collections.Generic.LinkedHashSet`1
+
+Please make sure that you have referenced the Iesi.Collections package.");
```

Comment:
For having it a bit more explicit, and letting users guess they are not more NHibernate features which may blow if not referencing Iesi:
`Please make sure that you have referenced the Iesi.Collections package, or do not use the order by feature on set mapping.`